great_expectations/profile/json_schema_profiler.py

```diff
@@ -215,15 +300,60 @@ def _create_set_expectation(
         self, key: str, details: dict
     ) -> Optional[ExpectationConfiguration]:
         """https://json-schema.org/understanding-json-schema/reference/generic.html#enumerated-values"""
-        if JsonSchemaTypes.ENUM.value not in details.keys():
+        object_types = self._get_object_types(details=details)
+
+        if JsonSchemaTypes.ENUM.value not in object_types:
             return None
+
         enum = details.get("enum", None)
-        if not isinstance(enum, list):
+        any_of = details.get("anyOf", None)
+
+        kwargs = {"column": key}
+
+        if enum:
+            kwargs["value_set"] = enum
+        elif any_of:
+            for item in any_of:
+                item_type = item.get("enum", None)
+                if item_type:
+                    kwargs["value_set"] = item_type
+                    break
+        else:
+            return None
+
+        if "value_set" not in kwargs:
+            return None
+
+        if not isinstance(kwargs["value_set"], list):
             return None
 
-        kwargs = {"column": key, "value_set": enum}
         return ExpectationConfiguration("expect_column_values_to_be_in_set", kwargs)
 
+    def _create_null_or_not_null_column_expectation(
+        self, key: str, details: dict
+    ) -> Optional[ExpectationConfiguration]:
+        """https://json-schema.org/understanding-json-schema/reference/null.html"""
+        object_types = self._get_object_types(details=details)
+        enum_ = details.get("enum", None)
+
+        kwargs = {"column": key}
+        null_expectation = ExpectationConfiguration(
+            "expect_column_values_to_be_null", kwargs
+        )
+        not_null_expectation = ExpectationConfiguration(
+            "expect_column_values_to_not_be_null", kwargs
+        )
+
+        if JsonSchemaTypes.NULL.value in object_types:
+            if len(object_types) == 1:
+                if enum_:
+                    return None
+                return null_expectation
+            else:
+                return None
+        else:
+            return not_null_expectation
```

Comment:
Inverting the nested logic into guard clauses makes this easier to read.

```suggestion
        if JsonSchemaTypes.NULL.value not in object_types:
            return ExpectationConfiguration("expect_column_values_to_not_be_null", kwargs)

        if len(object_types) != 1:
            return None

        if enum_:
            return None

        return ExpectationConfiguration("expect_column_values_to_be_null", kwargs)
```

Reply:
Inverted this and managed to clean it up better. Discovered enums and their pains and fixed the test cases and code to cover all the ways they can be null.